feat: Automatically enable Helm v3 mode

drakedevel opened the pull request:
Runs `helm version` in helmexec.New, and exposes a method on Interface to allow other packages to use the detected version. Preserves compatibility with previous HELMFILE_HELM3 mechanism.

Resolves #923

drakedevel commented:
Whoops, that CI failure is real -- the `TestHelmState_flagsForUpgrade` passes in a `ShellRunner` and tries to execute the system `helm` (which worked on my system but thankfully failed in CI). I'll pull in `mockRunner` for that test first thing tomorrow.

mumoshu commented on .circleci/config.yml:
```diff
@@ -136,7 +136,7 @@ jobs:
         name: Install helm
         environment:
-          HELM_VERSION: v3.0.0-rc.2
+          HELM_VERSION: v3.0.0
```
I think this resulted in the integration test to fail:

![image](https://user-images.githubusercontent.com/22009/68856405-c1821900-0723-11ea-9591-4b7cb9645835.png)

Actually, I've also encountered this while bumping this along with the helm version in Dockerfile.

The fix was https://github.com/roboll/helmfile/pull/965/files#diff-452b35dd32cfe6d2251b4e8a4f7e1fc0R49

drakedevel replied:
Ah, thanks for the fix!

The integration test actually failed once before I made that change -- I tried bumping to 3.0.0 but was nearly midnight and I missed quite a few places :stuck_out_tongue: I'll rebase onto #965.

drakedevel commented:
@mumoshu all green! I rebased on top of the un-merged #965; the changes from that appear here since that PR hasn't been merged yet.

I also added a tiny test for the Helm 3 detection logic for good measure.

I did want to quickly call one thing out about this change: as we saw in the integration test failures, this PR makes helmfile blow up on Helm 3 <3.0.0-rc.3. Is that OK? Confidently detecting the versions in the middle is possible (although pretty ugly), but I was thinking that since all affected versions are weeks-old prereleases it's probably better to just tell folks to upgrade to a released version.

mumoshu commented:
> this PR makes helmfile blow up on Helm 3 <3.0.0-rc.3. Is that OK?

@drakedevel Good point! I think that's okay.

Btw, just to be clear, does it fail on rc.2 and lesser? I was thinking that you've already done a great job on keeping the check for `HELMFILE_HELM3` at https://github.com/roboll/helmfile/pull/963/files#diff-766154150f2383ce34784cd65a8f9862R51 which keeps rc.2 and lesser working.

But again, this PR looks great regardless of it actually works for rc.2 and lesser.

Thanks a lot for your contribution!

drakedevel commented:
> Btw, just to be clear, does it fail on rc.2 and lesser?

@mumoshu that's right -- but only if you don't specify `HELMFILE_HELM3`  (since that bypasses the `version` check completely). So I guess that's actually less of a big deal -- you had to already been using that env var, so upgrading won't actually break anything.

> Thanks a lot for your contribution!
My pleasure, thanks for the fast reviews!
